### ISSUE TITLE
 Exception: Failed to find libgfortran.3.dylib

### ISSUE THREAD

**Issue by Karrenbelt** [eth-cscs/abcpy#43]
Hi, 

I was interested in trying out the abcpy, but run into an error when trying to pip install, which I wanted to share with you. I’m using macOS High Siera (10.13.6). The output I receive you find underneath. I also ran a locate to see if I could find the dynamic library, which I did, and also attached the output displaying this result.

best regards,
Michiel

```
(venv) Zarathustra:allosteric-inference-master Zarathustra$ sudo -H pip3 install abcpy
Password:
Collecting abcpy
  Downloading https://files.pythonhosted.org/packages/10/db/b399869792c81bb6046a43282c9cfa8be33db52d1b1876a0eb93a82cb38d/abcpy-0.5.2-py3-none-any.whl (83kB)
    100% |████████████████████████████████| 92kB 2.7MB/s 
Collecting coverage (from abcpy)
  Downloading https://files.pythonhosted.org/packages/a3/7e/c94c21d643bfe7017615994df7b52292a33c8dcf36a6f694af110594edba/coverage-4.5.1-cp36-cp36m-macosx_10_12_x86_64.whl (178kB)
    100% |████████████████████████████████| 184kB 10.4MB/s 
Requirement already satisfied: scipy in ./venv/lib/python3.6/site-packages (from abcpy) (1.1.0)
Collecting sphinx-rtd-theme (from abcpy)
  Downloading https://files.pythonhosted.org/packages/87/30/7460f7b77b6e8a080dd3688f750fe5d5666c49358f8941449c5b128fa97d/sphinx_rtd_theme-0.4.1-py2.py3-none-any.whl (5.4MB)
    100% |████████████████████████████████| 5.4MB 3.6MB/s 
Collecting sklearn (from abcpy)
  Downloading https://files.pythonhosted.org/packages/1e/7a/dbb3be0ce9bd5c8b7e3d87328e79063f8b263b2b1bfa4774cb1147bfcd3f/sklearn-0.0.tar.gz
Collecting sphinx==1.4.8 (from abcpy)
  Downloading https://files.pythonhosted.org/packages/b5/db/a93672c16532ee4066cebe47f9a72e774c9be44b976110a7a16cecc016fd/Sphinx-1.4.8-py2.py3-none-any.whl (1.6MB)
    100% |████████████████████████████████| 1.6MB 6.7MB/s 
Collecting glmnet==2.0.0 (from abcpy)
  Downloading https://files.pythonhosted.org/packages/c7/97/6f92f20fc193478c5d5927396c8d691abbdaa7774fd67e8a08fdeb1a2470/glmnet-2.0.0.tar.gz (102kB)
    100% |████████████████████████████████| 112kB 16.7MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-install-1bcfpco3/glmnet/setup.py", line 38, in <module>
        GFORTRAN_LIB = get_lib_dir('libgfortran.3.dylib')
      File "/private/tmp/pip-install-1bcfpco3/glmnet/setup.py", line 30, in get_lib_dir
        raise Exception("Failed to find {}".format(dylib))
    Exception: Failed to find libgfortran.3.dylib
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/tmp/pip-install-1bcfpco3/glmnet/
```

```
(venv) Zarathustra:allosteric-inference-master Zarathustra$ locate libgfortran.3.dylib
/Applications/MATLAB_R2017b.app/sys/os/maci64/libgfortran.3.dylib
/Applications/Tellurium.app/Contents/Resources/telocal/python-3.6.3/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
/Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libgfortran.3.dylib
/Users/Zarathustra/ETH/git_repo/allosteric-inference-master/venv/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
/Users/Zarathustra/ETH/repositories/allosteric-inference-master/python_server/tutorial/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
/Users/Zarathustra/ETH/repositories/allosteric-inference-master/venv/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
/Users/Zarathustra/ETH/repositories/allosteric-inference-master/venv/python_server/tutorial/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
/Users/Zarathustra/miniconda2/lib/libgfortran.3.dylib
/Users/Zarathustra/miniconda2/pkgs/libgfortran-3.0.1-h93005f0_2/lib/libgfortran.3.dylib
/Users/Zarathustra/miniconda3/lib/libgfortran.3.dylib
/Users/Zarathustra/miniconda3/pkgs/libgfortran-3.0.1-h93005f0_2/lib/libgfortran.3.dylib
/usr/local/lib/python3.6/site-packages/scipy/.dylibs/libgfortran.3.dylib
```

**Comment by Karrenbelt** [comment]
Thanks for the suggestion, that lead to a solution. I tried to install gfortran, which failed because it’s part of gcc nowadays, so I tried gcc (bit surprised I didn’t install it on this system yet), and that gave me the following:

xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
Error: Failure while executing; `git config --local --replace-all homebrew.private true` exited with 1.

Leading to the solution: https://tips.tutorialhorizon.com/2015/10/01/xcrun-error-invalid-active-developer-path-library-developer-commandline-tools-missing-xcrun/ <https://tips.tutorialhorizon.com/2015/10/01/xcrun-error-invalid-active-developer-path-library-developer-commandline-tools-missing-xcrun/> 
After running the recommended xcode-select --install, I brew installed gcc and pip3 installed abcpy without issues, and managed to import it into python.

Best regards,
Michiel
